kong/plugins/proxy-cache/handler.lua

```diff
@@ -163,25 +163,29 @@ local function cacheable_response(conf, cc)
   end
 
   do
-    local content_type = ngx.var.sent_http_content_type
+    local status = kong.response.get_status()
+    -- conform to RFC since 204/304 do not have Content-Type header
+    if status ~= 204 and status ~= 304 then
```

Comment:
It seems this change is not doing what you expect. If the response has status 204 or 304 you are saying that it should not have the content-type evaluated for caching. So, instead of cleaning the header that should not be sent, you are refusing to check if the content should be cached.

Also, if your intention was to avoid the caching at all, this is not right. The logic test here will only bypass the content-type evaluation, the function will continue to execute, will test the cache-control conditions and the response possibly will still be cached.

Reply:
We want that 204 and 304 are catched which is not the case without the patch.
We are talking that the plugin is expecting the header content type to cache the answer which is wrong